lib/ococo/meta/validate_params.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+const _isFinite = require('lodash/isFinite')
+
+const ORDER_TYPES = ['MARKET', 'LIMIT']
+
+module.exports = (args = {}) => {
+  const {
+    orderPrice, amount, orderType, submitDelay, cancelDelay, limitPrice,
+    stopPrice, ocoAmount, lev, _futures
+  } = args
+
+  if (ORDER_TYPES.indexOf(orderType) === -1) return `Invalid order type: ${orderType}`
+  if (!_isFinite(amount)) return 'Invalid amount'
+  if (!_isFinite(submitDelay) || submitDelay < 0) return 'Invalid submit delay'
+  if (!_isFinite(cancelDelay) || cancelDelay < 0) return 'Invalid cancel delay'
+  if (orderType === 'LIMIT' && !_isFinite(orderPrice)) {
+    return 'Limit price required for LIMIT order type'
+  }
+
+  if (!_isFinite(limitPrice)) return 'Invalid OCO limit price'
+  if (!_isFinite(stopPrice)) return 'Invalid OCO stop price'
```

Comment:
We could also check that OCO amount < original order amount otherwise it might be rejected for `insufficient balance`

Reply:
I wouldn't restrict it; if the order fails due to insufficient balance it automatically cancels, and it should be up to the user.